packages/typespec-go/src/tcgcadapter/clients.ts

```diff
@@ -171,7 +173,7 @@ export class clientAdapter {
   private adaptURIParam(sdkParam: tcgc.SdkPathParameter): go.URIParameter {
     const paramType = this.ta.getPossibleType(sdkParam.type, true, false);
     if (!go.isConstantType(paramType) && !go.isPrimitiveType(paramType)) {
-      throw new Error(`unexpected URI parameter type ${go.getTypeDeclaration(paramType)}`);
+      throw new AdapterError('UnsupportedTsp', `unsupported URI parameter type ${go.getTypeDeclaration(paramType)}`, sdkParam.__raw!.node);
```

Comment:
There are quite a few places where `__raw` is `?` which is unfortunate. @tadelesh are there cases where `__raw` and/or `__raw.node` will be absent? If so, then this will need to be cleaned up. Or is there a better way to get the node? Having the node gives a better user experience as the compiler will output file and line number info as part of the diagnostic.

Reply:
some types are not defined by user, but created by tcgc. for example, discriminated property of implicit `@discriminator`. for such cases, we do not have `__raw` type.

Reply:
fixed